R/data-storage.R

```diff
@@ -160,12 +164,25 @@ DataStorage <- R6::R6Class( # nolint object_name_linter
             #  empty strings.
             tmp_result <- .x  %>%
               jsonlite::fromJSON() %>%
-              purrr::compact() %>%
-              as.data.frame()  %>%
+              purrr::compact()
+
+            tmp_result <- tmp_result %>%
+              purrr::map(function(.x2) {
+                if (checkmate::test_atomic_vector(.x2, min.len = 2)) {
+                  return(list(.x2))
+                }
+                if (length(.x2) > 1) {
+                  return(jsonlite::toJSON(.x2, auto_unbox = TRUE))
+                }
+                .x2
+              })
+
+            tmp_result <- do.call(cbind, tmp_result) %>%
+              as.data.frame() %>%
```

Comment:
Can we use `purrr::list_cbind` here?
```suggestion
            tmp_result <- tmp_result %>%
              purrr::list_cbind() %>%
```

Reply:
It doesn't work 😕

This is a trick for {telemetry} to be able to convert a list where elements can have any type into a column-based data.frame with 1 row.

The output needs to preserve atomic elements and non-atomic as a "cell", not expanding to how many elements the object has.

```R
r> a_list
$id
[1] "vector_selected"

$value
$value[[1]]
 [1]  1  2  3  4  5  6  7  8  9 10


$custom
[1] 2

r> do.call(cbind, a_list)
     id                value      custom
[1,] "vector_selected" integer,10 2  
```